RadarSDK/Radar.m

```diff
@@ -65,19 +74,36 @@ + (void)initializeWithPublishableKey:(NSString *)publishableKey {
     [RadarLocationPermissionManager sharedInstance];
 
     [[RadarLocationManager sharedInstance] updateTrackingFromInitialize];
+    
     [[RadarAPIClient sharedInstance] getConfigForUsage:@"initialize"
                                               verified:NO
                                      completionHandler:^(RadarStatus status, RadarConfig *config) {
-                                         if (status != RadarStatusSuccess || !config) {
-                                            return;
+                                         if (status == RadarStatusSuccess && config) {
+                                             [[RadarLocationManager sharedInstance] updateTrackingFromMeta:config.meta];
+                                             [RadarSettings setFeatureSettings:config.meta.featureSettings];
+                                             [RadarSettings setSdkConfiguration:config.meta.sdkConfiguration];
+                                         }
+                                         
+                                         RadarSdkConfiguration *sdkConfiguration = [RadarSettings sdkConfiguration];
+                                         if (sdkConfiguration.startTrackingOnInitialize && ![RadarSettings tracking]) {
+                                            [Radar startTrackingWithOptions:[RadarSettings trackingOptions]];
+                                         }
+                                         if (sdkConfiguration.trackOnceOnInitialize) {
+                                            [Radar trackOnceWithCompletionHandler:nil];
                                          }
-                                         [[RadarLocationManager sharedInstance] updateTrackingFromMeta:config.meta];
-                                         [RadarSettings setFeatureSettings:config.meta.featureSettings];
+
                                          [self flushLogs];
                                      }];
-    
+    if (options.userId != nil) {
```

Comment:
Is there a reason we are setting the user ID and meta data after the initial sync? Do we want to send that information up to the server during initialize right away?

Reply:
I don't think those get sent during initialize, normally in code they are set after the init call, I just placed them at the end for convenience, we can move them up in case they are being used.

Reply:
Hmm its my impression that one of the intention of one line implementation is to automatically sync meta-data both ways during initialization time to reduce the lines of code that is required to integrate our SDK. To me it feels like an omission to not sync the metadata and userid provided directly to the client to the server.

Reply:
server doesn't not need to know about metadata/userId, it is passed up per track call.